### ISSUE TITLE
Lint highlighter issue in gutter

### ISSUE THREAD

**Issue by cideM** [mawww/kakoune#2332]
### Steps
Create a file that has a linter error. Run `:lint` so that a red square is shown at the start of the offending line. Modify that line in any way, e.g. by inserting a character. (I can reproduce this both in a rust and in a JS file)

### Outcome 
Red square vanishes (bad), the error message popup stays (good)

### Expected
Red square stays unless the error was fixed

**Comment by lenormf** [comment]
Looks like the `line-specs` highlighter updates the specs list whenever lines are modified, which explains why the flag disappears as soon as characters are inserted.

I think it makes sense to assume that a line should be flagged only as long as it remains the way it was when the flag was initialized.

As the editor has no way to know by itself whether a line is still technically the same (e.g. after you insert whitespace in javascript code) despite changes, you would probably need to run the linter on every key hit to have 100% accuracy, but that's not advisable obviously.

**Comment by cideM** [comment]
> I think it makes sense to assume that a line should be flagged only as long as it remains the way it was when the flag was initialized.

I'd say it makes sense to assume the state of the line hasn't changed until the linter is run again. Meaning a correct line shouldn't suddenly be flagged and an incorrect line shouldn't suddenly be *not flagged anymore* without input from the linter. In that sense I think the current behavior is not in fact what most people would expect but maybe that's just me.

**Comment by maximbaz** [comment]
I agree, line should remain marked until you run linter again. Same as git gutter, markers are present until you run `git show-diff` again.

**Comment by cideM** [comment]
Additionally, the current implementation means that you'll still have the popup window with the error message, but no line marker. So the current implementation is at least internally inconsistent.

**Comment by stevenyvr987** [comment]
It seems the lint gutter disappears after an insert, because the line-specs defining the gutter has been deleted by a buffer update.

To show this, I reduce `lint.kak` to the following `rc` file, which defines a gutter to show a red asterisk on the first line:
> declare-option -hidden line-specs lint_flags
add-highlighter window/lint flag-lines default lint_flags
update-option window lint_flags
set-option -add window lint_flags '1|{red}*'

I source the `rc` file on an input file which contains one line of text, and it readily shows the red gutter:
> src/kak.opt -n -e 'source test/lint/rc' test/lint/in

I echo `lint-flags` to the `*debug*` buffer before and after I insert a whitespace in the line:
> echo -debug %opt{lint_flags}

The gutter disappears as noted, and `*debug*` shows
>  1 1|{red}*
2

Update 1 shows the expected `lint_flags`; update 2 is after the insert and it shows empty `lint_flags`. It seems the disappearance of the lint gutter has nothing to do with the `lint.kak` code, but maybe with some unintended action between line-specs and buffer timestamps.
